docs/src/pages/components/data-grid/rows/InfiniteLoadingGrid.js

```diff
@@ -15,25 +17,36 @@ function CustomLoadingOverlay() {
 
 export default function InfiniteLoadingGrid() {
   const [loading, setLoading] = React.useState(false);
+  const timeout = React.useRef(null);
   const { data, setRowLength } = useDemoData({
     dataSet: 'Commodity',
     rowLength: 20,
     maxColumns: 6,
   });
 
   const loadServerRows = (newRowLength) => {
+    clearTimeout(timeout.current);
     setLoading(true);
-    setTimeout(() => {
+
+    timeout.current = setTimeout(() => {
       setRowLength(newRowLength);
       setLoading(false);
+      // Simulate network throttle
     }, Math.random() * 500 + 100);
   };
 
   const handleOnRowsScrollEnd = (params) => {
     const newRowLength = params.virtualRowsCount + params.viewportPageSize;
-    loadServerRows(newRowLength);
+
+    if (newRowLength <= MAX_ROW_LENGTH) {
+      loadServerRows(newRowLength);
+    }
   };
 
+  React.useEffect(() => {
+    return () => clearTimeout(timeout.current);
+  }, []);
+
   return (
     <div style={{ height: 400, width: '100%' }}>
       <XGrid
```

Comment:
there is also a rowLength prop that you can set the footer show the correct amount of rows and the scrollbar should size accordingly

Reply:
I don't see a `rowLength` prop on the grid. The footer is showing the total rows currently.

Reply:
Sorry
`rowCount?: number;`

Reply:
Ok to summarise our call from yesterday - I will follow up on part for the `rowCount` and the functionality where all the rows are allocated in advance and a callback is called requestion for a specific range of rows to be loaded. I still think that the feature as it now brings value and the extra mile of making it work in combination with `rowCount` can be done afterwards.

On another note - @dtassone we also discussed a way not to depend on the `scrollParams` that are being passed by hooking up directly to the state - the problem with this approach is that the state is updated asynchronously which means that by the time I need to check the scroll position it is not the correct one in the state. The way to fix it is to `forceUpdate` the state on each scroll, but that impacts the performance.  The overall conclusion is that it is better to just get the value from the `scrollParams` rather than from the state.

Reply:
why don't you pass the realScroll state in the event params. 
The event is fired from `updateViewport` where all the required params are, so you could just change the event payload and pass the real scroll params as well as the rendering scroll params.

Reply:
Ok to summarise our call from yesterday - I will follow up on part for the rowCount and the functionality where all the rows are allocated in advance and a callback is called requestion for a specific range of rows to be loaded. I still think that the feature as it now brings value and the extra mile of making it work in combination with rowCount can be done afterward.

PS: I removed my other comment -> result -> fixed.

Reply:
Much better without the absoluteTop 👍 . 
Ok, let's clean up this one a bit more and but scope this PR to the current feature. 
Please open an issue describing the rest of the feature with the rowCount. Those needs to work together.

Reply:
Here is the ticket -> https://github.com/mui-org/material-ui-x/issues/1247